site/app/controllers/UserProfileController.php

```diff
@@ -100,14 +99,38 @@ public function changePassword() {
     }
 
 
+    /**
+     * @Route("/user_profile/change_pronouns", methods={"POST"})
+     * @return JsonResponse
+     * @throws \ImagickException
+     */
+    public function changePronouns() {
+        $user = $this->core->getUser();
+        $newPronouns = trim($_POST['pronouns']);
+
+        //validPronouns() checks for valid option
+        if ($user->validateUserData('user_pronouns', $newPronouns) === true) {
+            $user->setPronouns($newPronouns);
+            $user->setUserUpdated(true);
+            $this->core->getQueries()->updateUser($user);
+            return JsonResponse::getSuccessResponse([
+                'message' => "Pronouns updated successfully",
+                'pronouns' => $newPronouns
+            ]);
+        }
+        else {
+            return JsonResponse::getErrorResponse("Pronouns is not valid");
+        }
+    }
+
     /**
      * @Route("/user_profile/change_preferred_names", methods={"POST"})
      * @return JsonResponse
      * @throws \ImagickException
      */
     public function changeUserName() {
         $user = $this->core->getUser();
-        if (empty($_POST['user_givenname_change']) && empty($_POST['user_familyname_change'])) {
+        if (!empty($_POST['given_name']) && !empty($_POST['family_name'])) {
```

Comment:
these are the bugs I mentioned in the meeting. 'user_' should be removed. That is what real POST variable names are supposed to be.

Reply:
I'm not sure I understand your response.  You need to first check to make sure the fields `given_name` and `family_name` exist in the array `$_POST` before you access them.  Otherwise, an exception will be thrown.

Reply:
oh I see. Fixed